### DISCUSSION TITLE
Adding a resource widget to an infolist viewentry/blade file

### DISCUSSION THREAD

**Discussion by gilbert-mojodojo** [filamentphp/filament#8605] [Help]
I created a resource widget which displays a barchart. It works as intended when using the getHeaderWidgets() function on the resource pages. I created an infolist on the resource entry's view page which uses a blade file to display some information. The problem is that if i were to use @livewire(\App\Filament\Resources\TestResource\Widgets\TestWidget::class) for example, it loads the widget but the widget no longer detects the current resource that is being displayed ($this->record is null). I was wondering if there's a way to make a call to the widget while maintaining the link to the current resource entry so I can display the barchart for whatever entry a user is viewing.

**Comment by leandrocfe** [comment]
Did you try to pass the param?
```php
@livewire(\App\Filament\Resources\TestResource\Widgets\TestWidget::class, ['record' => $record])
```

**Comment by dharen008** [comment]
by using the blade file
@livewire( \App\Filament\Resources\TestResource\Widgets\TestWidget::class, [ 'record' => $getRecord() ] )

if you want to use it on resource
Livewire::make( TestWidget::class ),

make sure inside the TestWidget you have this

use Illuminate\Database\Eloquent\Model;
public ?Model $record;
